libraries/botbuilder/tests/botFrameworkHttpClient.test.js

```diff
@@ -12,161 +17,204 @@ class TestBotFrameworkHttpClient extends BotFrameworkHttpClient {
+                'conversationId',
+                { type: 'message', conversation: {} }
+            );
+            assert.strictEqual(response.status, 404);
         });
 
-        it('should succeed to make call using override buildCredentials', async () => {
+        it('should succeed to make call using override buildCredentials', async function () {
+            nock('http://skillUrl').post('/api/good').reply(200, { id: 'some-id' });
 
-            nock('http://skillUrl')
-                .post('/api/good')
-                .reply(200, { id: 'some-id' });
-        
             const credentialProvider = new SimpleCredentialProvider('this-is-not-the-app-id-your-looking-for', '1');
             const client = new TestBotFrameworkHttpClient(credentialProvider, 'channels');
             const fromBotId = 'this-is-not-the-app-id-your-looking-for';
-            const response = await client.postActivity(fromBotId, 'toBotId', 'http://skillUrl/api/good', 'serviceUrl', 'conversationId', { type: 'message', conversation: { } });
-            strictEqual(response.status, 200);
+            const response = await client.postActivity(
+                fromBotId,
+                'toBotId',
+                'http://skillUrl/api/good',
+                'serviceUrl',
+                'conversationId',
+                { type: 'message', conversation: {} }
+            );
+            assert.strictEqual(response.status, 200);
         });
 
-        it('should fail to make call with wrong credentials', async () => {
-
-            nock('http://skillUrl')
-            .post('/api/good')
-            .reply(200, { id: 'some-id' });
+        it('should fail to make call with wrong credentials', async function () {
+            nock('http://skillUrl').post('/api/good').reply(200, { id: 'some-id' });
 
             const credentialProvider = new SimpleCredentialProvider('test-app-id', 'test-app-Secret');
-            const client = new TestBotFrameworkHttpClient(credentialProvider, 'channels');
-            try {
-                await client.postActivity('fromBotId', 'toBotId', 'http://skillUrl/api/good', 'serviceUrl', 'conversationId', { type: 'message', conversation: { } });
-            } catch(e) {
-                strictEqual(e.message, 'BotFrameworkHttpClient.postActivity(): Unable to get appCredentials to connect to the skill');
-            }
+            const client = new TestBotFrameworkHttpClientWithNoCredentials(credentialProvider, 'channels');
+
+            await assert.rejects(
+                client.postActivity(
+                    'fromBotId',
+                    'toBotId',
+                    'http://skillUrl/api/good',
+                    'serviceUrl',
+                    'conversationId',
+                    { type: 'message', conversation: {} }
+                ),
+                Error('BotFrameworkHttpClient.postActivity(): Unable to get appCredentials to connect to the skill')
+            );
         });
```

Comment:
I wanted to call this one out, specifically, because the original test passed, but it passed because the `await client.postActivity()` call doesn't actually throw. It's "supposed" to throw [here](https://github.com/microsoft/botbuilder-js/blob/main/libraries/botbuilder/src/botFrameworkHttpClient.ts#L72):

```js
const appCredentials = await this.getAppCredentials(fromBotId, toBotId);
if (!appCredentials) {
    throw new Error(
        'BotFrameworkHttpClient.postActivity(): Unable to get appCredentials to connect to the skill'
    );
}
```

but the call to `this.getAppCredentials()` always returns `MicrosoftAppCredentials('', '')` (with `TestBotFrameworkHttpClient` or some other `MicrosoftAppCredentials` object (normal BotFrameworkHttpClient). I'm not sure if this is intentional or not.

Reply:
A great example why `assert.rejects` is more straightforward than try/catch in tests.